config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java

```diff
@@ -313,35 +334,47 @@ private <T> T convert(String propertyName, Class<T> type, String value) {
         }
     }
 
-    private Optional<ConfigValue> findConfigValue(String propertyName) {
+    private Optional<ConfigValue> findConfigValue(String propertyName, Optional<String> profiledPropertyName) {
         for (ConfigSource source : sources) {
-            String value = source.getValue(propertyName);
+            String selectedProperty = null;
+            String value = null;
+            if (profiledPropertyName.isPresent()) {
+                // Try profiled property first
+                selectedProperty = profiledPropertyName.get();
+                value = source.getValue(profiledPropertyName.get());
+            }
+            if (value == null) {
+                selectedProperty = propertyName;
+                value = source.getValue(propertyName);
+            }
 
             if (null == value) {
                 // not in this one
                 continue;
             }
-
+            String rawValue = value;
+            String name = source.getName();
+            int ordinal = source.getOrdinal();
+            final String propName = selectedProperty;
```

Comment:
This name is not descriptive. Use `selectedPropertyFinal` or something similar, with a comment "required final variable, as it is used in lambdas"

Reply:
Done